addons/payment/controllers/portal.py

```diff
@@ -159,9 +159,18 @@ def payment_method(self, **kwargs):
         acquirers_sudo = request.env['payment.acquirer'].sudo()._get_compatible_acquirers(
             request.env.company.id, partner.id, force_tokenization=True, is_validation=True
         )
-        tokens = set(partner.payment_token_ids).union(
-            partner.commercial_partner_id.sudo().payment_token_ids
-        )  # Show all partner's tokens, regardless of which acquirer is available
+
+        # Get all partner's tokens, regardless of which acquirers are available and then
+        # filter tokens for disabled acquirers and remove duplicates.
+        # We use 'sudo' here to access tokens_ids regardless of the user logged in at the moment.
+        partner_tokens = (partner.payment_token_ids
+                          + partner.commercial_partner_id.sudo().payment_token_ids)
+        # There is no need of a 'sudoed' environments for the tokens, as the information is
+        # already contained in the partner_tokens and we are are only filtering data that
+        # we already got.
+        tokens = set(partner_tokens
+                     .filtered(lambda record: record.acquirer_id.state != 'disabled'))
+
```

Comment:
Still not the target of the current task AFAIK.
This diff should be removed from current PR (you already made a better version in #79253, and it'll be merged before).

Reply:
Thanks for the explanation, I did not understand before, hopefully it is as before I made  the changes.

Reply:
:+1: